src/modules/enrollment/components/EditableOccurrencePointValue.tsx

```diff
@@ -0,0 +1,63 @@
+import DateRangeIcon from '@mui/icons-material/DateRange';
+import EditIcon from '@mui/icons-material/Edit';
+import { ReactNode, useMemo } from 'react';
+import IconButtonContainer from './IconButtonContainer';
+import { useFormDialog } from '@/modules/form/hooks/useFormDialog';
+import { EnrollmentFieldsFragment, FormRole } from '@/types/gqlTypes';
+
+interface Props {
+  icon: 'calendar' | 'pencil';
+  enrollment: EnrollmentFieldsFragment;
+  formRole: FormRole;
+  children?: ReactNode;
+  title: string;
+}
+
+const EditableOccurrencePointValue = ({
+  children,
+  icon,
+  enrollment,
+  formRole,
+  title,
+}: Props) => {
+  const formDialogArgs = useMemo(
+    () => ({
+      title,
+      submitButtonText: 'Save',
+      pickListArgs: {
+        projectId: enrollment.project.id,
+        householdId: enrollment.householdId,
+      },
+    }),
+    [title, enrollment]
+  );
+
+  const { openFormDialog, renderFormDialog } = useFormDialog({
+    formRole,
+    record: enrollment,
+    localConstants: {
+      entryDate: enrollment.entryDate,
+      exitDate: enrollment.exitDate,
+    },
+  });
+  return (
+    <>
+      <IconButtonContainer
+        onClick={openFormDialog}
+        Icon={icon === 'calendar' ? DateRangeIcon : EditIcon}
+      >
+        {children}
+      </IconButtonContainer>
+      {renderFormDialog(formDialogArgs)}
+    </>
+  );
+};
+
+const PermissionWrappedEditableOccurrencePointValue = (props: Props) => {
```

Comment:
What do you think about trying to standardize on React.FC?
```suggestion
const PermissionWrappedEditableOccurrencePointValue = (props):React.FC<Props> => {
```

Reply:
I think I loosely would prefer not to, because 1) it doesn't accept generics, and 2) it feels less readable to me. Are there particular benefits that you like about it? Apparently it also adds a `children` prop implicitly (I had no idea) which could lead to un caught type errors ([link](https://juhanajauhiainen.com/posts/should-you-use-reactfc-to-type-your-components)). I don't feel super strongly about it though, it's not a big deal to treat generic components differently.

Reply:
Those are good points, there is some controversy with React.FC. I like it because I think it conveys intent well and I find the return type checking to be a helpful safeguard. But if you find it to be less readable, that is a strong point against it. I am in favor of standardizing how we declare components but it doesn't have to be with React.FC, I've just been using it for so long that it looks normal/default to me.

FWIW [I gather that as of Typescript 5.1 the `children` prop is no longer included implicitly](https://dev.to/asciibi/reactfc-is-fixed-5ckn) and the return type is more permissive.

Reply:
Oh neat, okay. Impossible to keep up with all the react updates 🏃‍♀️ 
Neither looks particularly normal to me, so, I'm happy to standardize around React.FC. I don't think its worth going back to change components, but going forward we can use React.FC for non generic components.

Reply:
Sounds great, thanks! Yeah I totally agree, it's not worth changing existing components. We can add it incrementally going forward. We probably should upgrade to typescript to 5.1 though, it shouldn't be too hard, I can tackle that if you want.

Lacking support for generic components is unfortunate; I suspect those are uncommon(?) in this project so it's probably okay

Reply:
Yeah go for it on the typescript upgrade! We dont have too many generic components. Just a few that are used a lot.